src/Serilog.Settings.Configuration/Settings/Configuration/ConfigurationReader.cs

```diff
@@ -221,6 +221,16 @@ void ApplyEnrichment(LoggerConfiguration loggerConfiguration, IReadOnlyDictionar
             IConfigurationArgumentValue GetArgumentValue(IConfigurationSection argumentSection)
             {
                 IConfigurationArgumentValue argumentValue;
+
+                // Reject configurations where an element has both scalar and complex
+                // values as a result of reading multiple configuration sources.
+                if (argumentSection.Value != null && argumentSection.GetChildren().Any())
+                    throw new InvalidOperationException(
+                        $"The value for the argument '{argumentSection.Path}' is assigned different value " +
```

Comment:
I keep thinking `ArgumentException` but I guess that's best reserved for actual method arguments.

Reply:
That was my first inclination, too, but then I noticed IOE is used elsewhere in the code for similar "processing" errors.